Python/pylifecycle.c

```diff
@@ -1296,43 +1296,33 @@ init_interp_main(PyThreadState *tstate)
 #endif
     }
 
-    // Turn on experimental tier 2 (uops-based) optimizer
-    // This is also needed when the JIT is enabled
+// Turn on experimental tier 2 (uops-based) optimizer
+// This is also needed when the JIT is enabled
 #ifdef _Py_TIER2
     if (is_main_interp) {
-        int enabled = 1;
-#if _Py_TIER2 & 2
-        enabled = 0;
-#endif
-        char *env = Py_GETENV("PYTHON_JIT");
-        if (env && *env != '\0') {
-            // PYTHON_JIT=0|1 overrides the default
-            enabled = *env != '0';
+// perf profiler works fine with tier 2 interpreter, so
+// only checking for a "real JIT".
+#if _Py_TIER % 2 != 0
+        if (_PySys_JITEnabled() && config->perf_profiling > 0) {
+            (void)PyErr_WarnEx(
+                PyExc_RuntimeWarning,
+	            "JIT deactivated as perf profiling support is active",
+	            0);
         }
-        if (enabled) {
-#ifdef _Py_JIT
-            // perf profiler works fine with tier 2 interpreter, so
-            // only checking for a "real JIT".
-            if (config->perf_profiling > 0) {
-                (void)PyErr_WarnEx(
-                    PyExc_RuntimeWarning,
-                    "JIT deactivated as perf profiling support is active",
-                    0);
-            } else
+        else
 #endif
-            {
-                PyObject *opt = _PyOptimizer_NewUOpOptimizer();
-                if (opt == NULL) {
-                    return _PyStatus_ERR("can't initialize optimizer");
-                }
-                if (_Py_SetTier2Optimizer((_PyOptimizerObject *)opt)) {
-                    return _PyStatus_ERR("can't install optimizer");
-                }
-                Py_DECREF(opt);
-            }
+    {
+        PyObject *opt = _PyOptimizer_NewUOpOptimizer();
+        if (opt == NULL) {
+            return _PyStatus_ERR("can't initialize optimizer");
+        }
+        if (_Py_SetTier2Optimizer((_PyOptimizerObject *)opt)) {
+            return _PyStatus_ERR("can't install optimizer");
         }
+        Py_DECREF(opt);
     }
-#endif
+}
```

Comment:
This may require an indentation here (because it's the end of the if-block with `is_main_interp`).

Reply:
I hope that I understand your point correctly. Please check the last commit.

Reply:
Yes it's what I had in mind. That way, people reading the code won't have blocks that are not indented as intended. More generally, I think indentation (whenever possible) should be *as if* there was no `#if`-guard.

Simply put, I just assume that all `#if` guards are true and then I indent the code normally.